### PR TITLE
Set canonical link to original video/playlist url

### PR DESCRIPTION
**Pull request by tcitworld**
The local URL was used instead of the original url

## Description

The local URL was used instead of the original url

## Related issues

Previously fixed in https://github.com/Chocobozzz/PeerTube/issues/1539

Added a test to check but didn't manage to run them, so they're all yours.

### CODE REVIEW

File: server/tests/client.ts [hunk 1]
```diff
@@ -47,8 +51,19 @@ describe('Test a client controllers', function () {
   before(async function () {
     this.timeout(120000)
 
-    server = await flushAndRunServer(1)
-    server.accessToken = await serverLogin(server)
+    servers = await flushAndRunMultipleServers(2)
```
Comment (kontrollanten) [inline server/tests/client.ts:54]: A fancy way would be `[server, secondServer] = await flushAndRunMultipleServers(2)`

File: server/tests/client.ts [hunk 2]
```diff
@@ -47,8 +51,19 @@ describe('Test a client controllers', function () {
...
+    const server = servers[0]
+
+    await setAccessTokensToServers(servers)
+
+    {
+      const res = await uploadVideo(servers[0].url, servers[0].accessToken, { name: 'video' })
```
Comment (kontrollanten) [inline server/tests/client.ts:60]: Do we really need to upload a video here? According to line 78 there's already a video on the server.

If we need this line, then line 78 has to be updated (that's why tests are failing).

File: server/tests/client.ts [hunk 3]
```diff
@@ -318,6 +333,11 @@ describe('Test a client controllers', function () {
 
       checkIndexTags(res.text, 'PeerTube updated', 'my short description', 'body { background-color: red; }')
     })
+
+    it('Should use the original video URL for the canonical tag', async function () {
+      const res = await makeHTMLRequest(servers[1].url, '/videos/watch/' + videoUUID)
+      expect(res.text).to.contain(`<link rel="canonical" href="${videoOriginalUrl}" />`)
```
Comment (kontrollanten) [inline server/tests/client.ts:339]: By using `server.video` we could change the test to:

```
      const res = await makeHTMLRequest(servers[1].url, '/videos/watch/' + server.video.uuid)
      expect(res.text).to.contain(`<link rel="canonical" href="${server.video.url}" />`)
```

**Review by kontrollanten — CHANGES_REQUESTED** [review]

**Comment by Chocobozzz** [comment]
Thanks @tcitworld  @kontrollanten

**Pull request merged**